atmcfg/atmcfg.go

```diff
@@ -72,9 +77,78 @@ func setDisabledByShardName(out *opsmngr.AutomationConfig, name string, disabled
 	}
 }
 
+func setDisableProcessByProcessNameAndPort(out *opsmngr.AutomationConfig, clusterName string, processesMap map[string]bool, disabled bool) {
+	newDeploymentAuthMechanisms(out)
+	for k, p := range out.Processes {
+		key := fmt.Sprintf("%s:%d", p.Hostname, p.Args26.NET.Port)
+		if _, ok := processesMap[key]; ok && strings.Contains(out.Processes[k].Name, clusterName) {
```

Comment:
[nit] this is wrong, names can be changed by users, relying on our built in name generator (clustername+number) will make this fail if users customize it, you can't trust this will work for all, this is even worst for sharded cluster and will never work for them, if a user gives a sharded cluster name here you will never find the process

Reply:
I was erroneously assuming that the process name would change in case the user renames the cluster